src/modules/constructRelationalGraph.js

```diff
@@ -0,0 +1,265 @@
+      isValidJoinColumn(selection.rhs, tables)
+    ) {
+      const {lhs, rhs} = selection;
+      const leftTable = getTableFromAttribute(lhs, tables);
+      const rightTable = getTableFromAttribute(rhs, tables);
+      addEdge(graph, leftTable, rightTable, selection);
+    } else {
+      const {lhs, rhs} = selection;
+      const leftTable = getTableFromAttribute(lhs, tables);
+      const rightTable = getTableFromAttribute(rhs, tables);
+      const table = leftTable || rightTable;
+      if (table !== '') addNode(graph, table, expr);
+      else globalSelections.push(expr);
+    }
+  } else {
+    console.error(
+      'Invalid selection expression for Join Order Optimization',
+      expr
+    );
+  }
+};
+
+const parseJoinExpression = (
+  graph: Graph,
+  expr: Expression,
+  tables: Array<string>
+) => {
+  if (ExpressionType.AND in expr) {
+    const conditions = expr[ExpressionType.AND].clauses;
+    for (const condition of conditions) {
+      parseJoinExpression(graph, condition, tables);
+    }
+  } else if (ExpressionType.CMP in expr) {
+    const {lhs, rhs} = expr[ExpressionType.CMP];
+    const leftTable = getTableFromAttribute(lhs, tables);
+    const rightTable = getTableFromAttribute(rhs, tables);
+    addEdge(graph, leftTable, rightTable, expr[ExpressionType.CMP]);
+  } else {
+    console.error('Invalid join expression for Join Order Optimization', expr);
+  }
+};
+
+// function used to check all the tables that are involved in this query
+const preParseExpression = (expr: Expression, tables: Array<string>) => {
+  if (typeof expr === 'object') {
+    if (ExpressionType.RELATION in expr)
+      tables.push(expr[ExpressionType.RELATION]);
+    for (const key in expr) {
+      preParseExpression(expr[key], tables);
+    }
+  } else if (Array.isArray(expr)) {
+    for (const item of expr) {
+      preParseExpression(item, tables);
+    }
+  }
+};
+
+const parseExpression = (
+  graph: Graph,
+  expr: Expression,
+  tables: Array<string>,
+  globalSelections: Array<SelectionCondition>
+): void => {
+  if (ExpressionType.SELECTION in expr) {
+    const selection = expr[ExpressionType.SELECTION].arguments.select;
+    parseSelectionExpression(graph, selection, tables, globalSelections);
+    const children = expr[ExpressionType.SELECTION].children;
+    for (const child of children) {
+      parseExpression(graph, child, tables, globalSelections);
+    }
+  } else if (ExpressionType.JOIN in expr) {
+    const joinExpr = expr[ExpressionType.JOIN];
+    const {left, right, condition} = joinExpr;
+    parseJoinExpression(graph, condition, tables);
+    parseExpression(graph, left, tables, globalSelections);
+    parseExpression(graph, right, tables, globalSelections);
+  } else if (ExpressionType.RELATION in expr) {
+    const table = expr[ExpressionType.RELATION];
+    addNode(graph, table);
+  } else {
+    console.error('Invalid expression type', expr);
+    throw new Error('Invalid expression type');
```

Comment:
Try to avoid both logging and throwing an error. (In general, throwing an error is preferred since we can catch that and do something with it.